client-cli/src/command/wallet_command.rs

```diff
@@ -56,17 +56,20 @@ impl WalletCommand {
                 "Passphrases do not match",
             ));
         }
+        let enckey = derive_enckey(&passphrase, name).err_kind(ErrorKind::InvalidInput, || {
+            "Unable to derive encryption key from passphrase"
+        })?;
 
-        let mnemonic = wallet_client.new_wallet(name, &passphrase, wallet_kind)?;
-
-        success(&format!("Wallet created with name: {}", name));
+        let mnemonic = wallet_client.new_wallet(name, &enckey, wallet_kind)?;
 
         if let WalletKind::HD = wallet_kind {
             ask("Please store following mnemonic safely to restore your wallet later: ");
             println!();
             success(&mnemonic.unwrap().unsecure_phrase());
         }
 
+        println!();
+        println!("{}", &hex::encode(enckey.unsecure()));
```

Comment:
Use `success()` instead of `println!()` here. Also add a message (something like: `"Here is your authentication token to be used in future client requests: {}"`)

Reply:
I use `println!` specifically here is to make the result parsable by CLI tools.

Reply:
How will the user know what this random string means? We should atleast print some message for him.

Reply:
Done, scripts can use regex to extract information.